src/core/types/capabilities.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * The read-only set of capabilities available for the current UI session.
+ * Capabilities are simple key-value pairs of (string, boolean), where the string denotes the capability ID,
+ * and the boolean is a flag indicating if the capability is enabled or disabled.
+ *
+ * @public
+ */
+export interface Capabilities {
```

Comment:
The exact same interface is used from both client and server. Previously, server was importing it from client. I though this was something we wanted to avoid, so to avoid duplicating the type, I moved it under `types`. From what I understand, josh was saying that was the purpose of this module. Should I do otherwise?

Reply:
We follow the convention that the server is a source of truth for common types. We use a separate  `types.ts` file in a domain folder for declarations. Example https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/types.ts

Hopefully, one day we can move to the real per-domain separation structure and place `public`, `server` folder under a domain folder. In this case, we can introduce the `common` sub-folder to keep all the code to share.

Reply:
`server` being the source of truth in absense of `common/shared` module(s) makes sense to me. However, with our current public/server separation, I'm quite afraid to use `server` as the default place for shared type, as you never know exactly what babel/webpack will import in the client bundles even with named imports.

For exemple, if I put `Capabilities` in `core/server/types` and also export it in `core/server/index.ts`, when importing from `core/server/types` from client, even my IDE warn me that the import can be shortened to `core/server`, which would result in importing unwanted things to client (even if tree-shaking should be a thing)

In addition, we discussed about https://github.com/elastic/kibana/issues/49053 which should forbid this (but that was me who opened in in my first weeks in the team so I may have misunderstood)

Not sure what to do then?

Reply:
Let's merge the PR and discuss it on the weekly. This topic pops up several times. I'm down for introducing the real per-domain separation for the core if there are no technical limitations.